### PR TITLE
Minimal dark mode

### PR DESCRIPTION
**Pull request by RobbiNespu**
Ref: https://github.com/RobbiNespu/archie/pull/4/files

Minimal dark mode archie #13

### CODE REVIEW

**Review by athul — CHANGES_REQUESTED** [review]
Hey, I tried this locally, but can't change to light mode when system theme is set to dark. Not what I had in mind though 😅 . Could you patch this to change the theme even when the system theme is set on one(dark/light)

Also remove the console logs 😄

**Comment by RobbiNespu** [comment]
Hi @athul , yes as mentioned on #13 , it for using light mode `manual` or light mode `auto`, the toggle will be visible. 

there is only 1 icon which is `fa-sun` for both. Even though the icon name "sun" , It look like lamp to me. If you check the setTheme() funtion, I try replace icon with `toggle.className` but seem it not work. (that is where I left console.log) , maybe you or anyone can take alook?

But nevertheless , what I need is archive but it not likely a properly work and what people expected.

Someone else need to take care it. I not good with JS. You can give me change suggestion if you found the answer or just edit the commit and make changes

File: layouts/partials/footer.html [hunk 1]
```diff
@@ -9,6 +9,10 @@
 {{ template "_internal/google_analytics_async.html" . }}
 {{ end }}
 
+{{- if or (eq .Site.Params.mode "auto") (eq .Site.Params.mode "light") -}}
+  <script src="{{ .Site.BaseURL }}js/toggle.dark.js"></script>
```
Comment (RobbiNespu) [inline layouts/partials/footer.html:13]: @athul maybe need to if else logic here. So if dark mode is there, then load toggle.light.js

File: layouts/partials/header.html [hunk 2]
```diff
@@ -43,6 +43,11 @@
 		{{ $darkstyle := resources.Get "css/dark.css" | fingerprint }}
 		<link rel="stylesheet" type="text/css" href="{{ $darkstyle.Permalink }}" {{ if eq .Site.Params.mode "auto" }}media="(prefers-color-scheme: dark)"{{ end }} />
 	{{ end }}
+
+	{{- if or (eq .Site.Params.mode "auto") (eq .Site.Params.mode "light") -}}
+		{{ $darkstyle := resources.Get "css/dark.css" | fingerprint }}
+		<link disabled id="dark-mode-theme" rel="stylesheet" href="{{ $darkstyle.Permalink }}">
```
Comment (RobbiNespu) [inline layouts/partials/header.html:49]: @athul this is where, it critical. if theme is not dark mode, we can load the dark.css, but if theme are dark should we load main.css? It sound weird to me

**Comment by athul** [comment]
Hey, @RobbiNespu sorry I couldn't look more into it. I really am. But had to merge #36 since it was more concise and lesser code. Hope you can understand. 

Thanks for the PR and wishing you good luck 🤝

**Comment by RobbiNespu** [comment]
Hello [@athul](https://github.com/athul) , it fine. I am okey to close the PR #30. I also agreed PR #36 it was more concise and lesser code, perhaps much cleaner. Thanks for mention and wishing you good luck too 🤝

(Originally published at: https://robbinespu.gitlab.io/indieweb/2021-09-29-0000-comment-github-archie/)

**Pull request closed without merging**